src/modules/kvs/lookup.c

```diff
@@ -606,16 +649,38 @@ bool lookup (lookup_t *lh)
                     lh->errnum = ENOTDIR;
                     goto done;
                 }
-                reftmp = json_string_value (vp);
+                if ((refcount = treeobj_get_count (lh->wdirent)) < 0) {
+                    lh->errnum = errno;
+                    goto done;
+                }
+                if (refcount != 1) {
+                    lh->errnum = EPERM;
```

Comment:
Yeah, EPERM was what I chose for all "internally invalid" kind of situations.  Maybe that wasn't the right one from the beginning.  But I think adding some log messages is a good idea.